aten/src/THCS/generic/THCSTensor.cpp

```diff
@@ -347,12 +371,27 @@ void THCSTensor_(copy)(THCState *state, THCSTensor *self, THCSTensor *src) {
   THCSTensor_(_set)(state, self, src->indices, src->values);
   self->nnz = src->nnz;
   self->coalesced = src->coalesced;
+  THCSTensor_(_set_csr)(state, self, src->csr);
 }
 
 int THCSTensor_(isCoalesced)(THCState *state, const THCSTensor *self) {
   return self->coalesced;
 }
 
+int THCSTensor_(hasCSR)(THCState *state, const THCSTensor *self) {
+  return THCudaIntTensor_nDimension(state, self->csr) != 0;
+}
```

Comment:
I think that since this is an optional and short-lived attribute, then we would be better off using a null pointer to represent a missing value.